pkg/sysregistriesv2/alias.go

```diff
@@ -0,0 +1,112 @@
+package sysregistriesv2
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/homedir"
+	"github.com/containers/storage/pkg/lockfile"
+)
+
+// userAliasesFile is the user-specific registries.conf file to store aliases.
+var userAliasesFile = filepath.FromSlash(".config/containers/aliases.conf")
+
+// userAliasesLockFile is the lockfile for the user-specific registries.conf
+// file to store aliases.
+var userAliasesLockFile = filepath.FromSlash(".local/share/containers/aliases.conf.lock")
+
+// ResolveAlias performs an alias resolution of the specified name.  The
+// user-specific aliases.conf has precedence over aliases in `conf`.
+func (conf *V2RegistriesConf) ResolveAlias(ctx *types.SystemContext, name string) (string, bool, error) {
```

Comment:
`V2RegistriesConf` is not a natural place to provide this; that type represents “a raw configuration file” basically for the purpose of `ConvertToV2`; there isn’t even a natural way for callers to _get_ a `V2RegistriesConf` value, and we don’t want them to think about individual files anyway.

I’d expect a top-level function similar to `FindRegistry`.

Also, I shouldn’t this use use the existing caching mechanism ? (I do feel fairly unsure about that, admittedly.)

Reply:
I felt into the trap while working `pkg/shortnames`.

> Also, I shouldn’t this use use the existing caching mechanism ? 

Absolutely. That's the beauty of having a separate `aliases.conf` - we don't invalidate the registries.confs when adding a new alias.

Reply:
I was actually thinking about using the caching mechanism for `aliases.conf` as well — at least a single Podman command probably doesn’t care to, or might even actively _not_ want to, notice the alias changing over a run.

OTOH it _might_ be possible, with interactive use and indefinite user-driven delays, for users to observe that an entry was updated in one terminal but not used in another. So I’m not at all sure. (Either way, the caching is not contractual and we can add/remove it later.)

CRI-O already has the `InvalidateCache` etc. mechanism, assuming it should even care.